earth_enterprise/src/fusion/autoingest/tools/geaddtoimageryproject.cpp

```diff
@@ -151,21 +173,39 @@ main(int argc, char *argv[]) {
         (mercator? kMercatorProjectSubtype : kProjectSubtype));
 
     // TimeMachine only applies to non-mercator
-     if (!mercator) {
+    if (!mercator) {
       req.enable_timemachine = enable_historical_imagery;
       req.disable_timemachine = disable_historical_imagery;
     }
 
+    // Only pass along the overlay-related options if it's a terrain project
+    if (AssetDefs::Terrain == AssetType){
+      req.enable_terrain_overlay = enable_terrain_overlay;
+      req.disable_terrain_overlay = disable_terrain_overlay;
+      req.overlay_terrain_start_level = start_level;
+      req.overlay_terrain_resources_min_level = resource_min_level;
+    }
+
     if (req.items.empty()) {
       if (cmd == &khAssetManagerProxy::RasterProjectNew ||
           cmd == &khAssetManagerProxy::MercatorRasterProjectNew) {
         notify(NFY_WARN,
                "No insets specified. Project will be empty.");
       } else if (mercator ||
-            !(enable_historical_imagery || disable_historical_imagery)) {
-        // For adding/modifying, we always need insets except:
-        // non-mercator when historical_imagery/no_historical_imagery is
-        // specified.
+        !( (AssetDefs::Imagery == AssetType && (enable_historical_imagery || disable_historical_imagery)) ||
+           (AssetDefs::Terrain == AssetType && 
+              (enable_terrain_overlay || disable_terrain_overlay || 
+              start_level != 0 || resource_min_level != 0))
+        )){
+            // !(AssetDefs::Imagery == AssetType && (enable_historical_imagery || disable_historical_imagery)) ||
```

Comment:
Is this is a dead code ? If yes let's remove it

Reply:
Yes, that's dead code. I'll remove it.